nemo/collections/tts/modules/ed_mel2spec.py

```diff
@@ -0,0 +1,264 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import math
+from collections import OrderedDict
+from enum import Enum
+
+import librosa
+import torch
+import torch.nn as nn
+from numpy.linalg import pinv
+
+from nemo.core.classes import Exportable, NeuralModule, typecheck
+from nemo.core.neural_types.elements import SpectrogramType
+from nemo.core.neural_types.neural_type import NeuralType
+from nemo.utils.decorators import experimental
+
+
+def str2act(txt):
+    """Translates text to neural network activation"""
+    return {
+        "sigmoid": nn.Sigmoid(),
+        "relu": nn.ReLU(),
+        "none": nn.Sequential(),
+        "lrelu": nn.LeakyReLU(0.2),
+        "selu": nn.SELU(),
+    }[txt.lower()]
+
+
+class OperationMode(Enum):
+    """Training or Inference (Evaluation) mode"""
+
+    training = 0
+    validation = 1
+    infer = 2
+
+
+def create_mel_filterbank(*args, **kwargs):
+    return librosa.filters.mel(*args, **kwargs)
+
+
+@experimental
+class EDMel2SpecModule(NeuralModule, Exportable):
+    def __init__(
+        self,
+        n_fft: int,
+        hop_length: int,
+        mel_fmin: int,
+        mel_fmax: int,
+        mel_freq: int,
+        layers: int,
+        sampling_rate: int,
+        widening: int,
+        use_batchnorm: bool,
+        droprate: float,
+        num_dropout: int,
+        pre_final_lin: bool,
+        act1: str,
+        act2: str,
+        use_weight_norm: bool,
+    ):
+        """
+        Encoder Decoder module trained to transfrom Mel encoded spectrograms to normal spectrograms
+
+        Args:
+
+        """
```

Comment:
Same comment as in the model. Can you describe this module better? Is it a reversible convolution stack?